homework3/hw3/Views/LeagueView.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  LeagueView.swift
+//  hw2
+//
+//  Created by Tina Jureško on 14.03.2025..
+//
+
+import Foundation
+import UIKit
+import SofaAcademic
+import SnapKit
+
+class LeagueView: BaseView {
+    
+    private let countryLabel = UILabel()
+    private let nameLabel = UILabel()
+    private let logoImageView = AsyncImageView()
+    private let arrowImageView = UIImageView()
+        
+    override func addViews() {
+        super.addViews()
+        addSubview(logoImageView)
+        addSubview(countryLabel)
+        addSubview(arrowImageView)
+        addSubview(nameLabel)
+    }
+
+    override func styleViews() {
+        arrowImageView.image = UIImage(named: "Vector")
+        
+        countryLabel.font = .regularBold14
+        countryLabel.textColor = .primaryBlack
+        countryLabel.numberOfLines = 0
+        countryLabel.lineBreakMode = .byWordWrapping
+        
+        nameLabel.font = .regularBold14
+        nameLabel.textColor = .secondaryGray
+        nameLabel.numberOfLines = 0
+        nameLabel.lineBreakMode = .byWordWrapping
+    }
+        
+    override func setupConstraints() {
+        super.setupConstraints()
+        
+        logoImageView.snp.makeConstraints {
+            $0.leading.equalToSuperview().offset(16)
+            $0.top.bottom.equalToSuperview().inset(12)
+            $0.size.equalTo(32)
+        }
+        
+        countryLabel.snp.makeConstraints {
+            $0.centerY.equalTo(logoImageView)
+            $0.leading.equalTo(logoImageView.snp.trailing).offset(32)
+            $0.width.lessThanOrEqualTo(200)
```

Comment:
Dobro si se snašla s lessThanOrEqualTo, ali čim se tu idu neki magic numberi definirati znači da nešto nije dobro. 
Možeš ovdje staviti maxWidth.

Ili definirati prioritete ako countryName i leagueName postanu jako dugački, tko će se prvi raditi kompresiju, odnosno koji se stišće koji širi.

https://stackoverflow.com/questions/15850417/cocoa-autolayout-content-hugging-vs-content-compression-resistance-priority

To malo proči, pa primjene na nekoj od sljedećih zadaća.